src/hooks/services/useFetchData.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+
+type FetchProps = {
+  url: string;
+};
+
+type ErrorType = {
+  message: string;
+  status: boolean;
+};
+
+const ErrorInitialState: ErrorType = {
+  message: "",
+  status: false,
+};
+
+function useFetchData<T>({ url }: FetchProps) {
+  const [data, setData] = useState<T[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState<ErrorType>(ErrorInitialState);
+
+  useEffect(() => {
+    const RequestData = async () => {
+      setIsLoading(true);
+      setHasError(ErrorInitialState);
+      try {
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error(
+            `An error has occured. Error status: ${response.status}`
+          );
+        }
+        const data = await response.json();
+        setData(data);
+        setIsLoading(false);
+      } catch (error) {
+        setHasError({ message: (error as Error).message, status: true });
+        setIsLoading(false);
+      }
+    };
+
+    RequestData();
+  }, [url]);
```

Comment:
i mostly added the url there because my linter was crying: `React Hook useEffect has a missing dependency: 'url'. Either include it or remove the dependency array.`

Now on the actual issue.
1. No definitely not. also ive noticed that whenever the page that calls the hook mounts, it fetches again which is horrible for the network optimization and our poor backend. So this needs some kind of caching. Can this be solved without Redux/Context?
2. Also nope. the thing is i pass it as an object which is a reference type and i think those should be considered as new instances for each rerender, so this could have some implications. tested it and it seems it doesnt behave that way, we could discuss this.

Also with an empty dep array, what happens if the request fails? it never fires again.